pkg/runner/writer_json.go

```diff
@@ -0,0 +1,35 @@
+package runner
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"github.com/linuxsuren/api-testing/pkg/apispec"
+	"io"
+)
+
+type jsonResultWriter struct {
+	writer       io.Writer
+	apiConverage apispec.APIConverage
+}
+
+// NewJSONResultWriter creates a new jsonResultWriter
+func NewJSONResultWriter(writer io.Writer) ReportResultWriter {
+	return &jsonResultWriter{writer: writer}
+}
+
+// Output writes the JSON base report to target writer
+func (w *jsonResultWriter) Output(result []ReportResult) (err error) {
+	jsonData, err := json.Marshal(result)
+	if err != nil {
+		return err
+	}
+	fmt.Fprint(w.writer, string(jsonData))
+	return nil
+}
+
+// WithAPIConverage sets the api coverage
+func (w *jsonResultWriter) WithAPIConverage(apiConverage apispec.APIConverage) ReportResultWriter {
+	w.apiConverage = apiConverage
```

Comment:
You didn't use `apiConverage` in this case. Making it to be empty is pretty reasonable.

Reply:
Allowing a function to do something that doesn't match its name will be confusing, and I don't think it would hurt to have it here.

Reply:
Of course, this is not a big problem. But having an empty function is quite normal. This could be an optional suggestion. I don't block the process because of this comment.